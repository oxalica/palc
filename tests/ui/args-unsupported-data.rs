use palc::{__private::Args, Args};

#[derive(Args)]
struct Unit;

#[derive(Args)]
struct Tuple();

#[derive(Args)]
union Union {
    a: (),
}

#[derive(Args)]
enum Enum {}

struct AssertImplArgs
where
    Unit: Args,
    Tuple: Args,
    Enum: Args;

fn main() {}
