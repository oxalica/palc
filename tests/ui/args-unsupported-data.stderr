error: only structs with named fields are supported
 --> tests/ui/args-unsupported-data.rs:3:10
  |
3 | #[derive(Args)]
  |          ^^^^
  |
  = note: this error originates in the derive macro `Args` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only structs with named fields are supported
 --> tests/ui/args-unsupported-data.rs:6:10
  |
6 | #[derive(Args)]
  |          ^^^^
  |
  = note: this error originates in the derive macro `Args` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only structs with named fields are supported
 --> tests/ui/args-unsupported-data.rs:9:10
  |
9 | #[derive(Args)]
  |          ^^^^
  |
  = note: this error originates in the derive macro `Args` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only structs with named fields are supported
  --> tests/ui/args-unsupported-data.rs:14:10
   |
14 | #[derive(Args)]
   |          ^^^^
   |
   = note: this error originates in the derive macro `Args` (in Nightly builds, run with -Z macro-backtrace for more info)
